### ISSUE TITLE
Implement `process.dlopen`

### ISSUE THREAD

**Issue by sno2** [oven-sh/bun#903]
### What is the problem this feature will solve?

This will help build the bridge of interoperability between Bun and Node. I haven't found anything that is blocked by this as it was added in Node 15 so it is most likely low priority.

### What is the feature you are proposing to solve the problem?

Implement https://nodejs.org/dist/latest-v18.x/docs/api/process.html#processdlopenmodule-filename-flags

**Comment by sno2** [comment]
Hmm it was actually implemented just not in the types, I will fix.
